File: src/main/java/com/atlaspharmacy/atlaspharmacy/users/domain/PharmacyAdmin.java
package com.atlaspharmacy.atlaspharmacy.users.domain;

import com.atlaspharmacy.atlaspharmacy.generalities.domain.Address;
import com.atlaspharmacy.atlaspharmacy.pharmacy.domain.Pharmacy;
import com.atlaspharmacy.atlaspharmacy.users.domain.Authority;
import com.atlaspharmacy.atlaspharmacy.users.domain.User;
import com.atlaspharmacy.atlaspharmacy.users.domain.enums.Role;
import org.hibernate.annotations.Proxy;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "pharmacyAdmins")
@DiscriminatorValue(value = Role.Values.PharmacyAdmin)
public class PharmacyAdmin extends User {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    private Pharmacy pharmacy;


    public PharmacyAdmin(Long id, Pharmacy pharmacy, boolean enabled, String verification_code) {
        this.id = id;
        this.pharmacy = pharmacy;

    }

    public PharmacyAdmin(){};


    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }


    public Pharmacy getPharmacy() {
        return pharmacy;
    }

    public void setPharmacy(Pharmacy pharmacy) {
        this.pharmacy = pharmacy;
    }

    @Override
    public String getUsername() {
        return getEmail();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }



}

File: frontend/src/app/model/pharmacyAdmin/pharmacyAdmin.ts
import {Gender} from "@app/model/patient/gender";
import {Address} from "@app/model/address/address";
import {Role} from "@app/model/users/role";

export class PharmacyAdmin{
  public name : String;
  public surname : String;
  public dateOfBirth : Date;
  public phoneNumber : String;
  public email : String;
  public gender : Gender;
  public address : Address;

  constructor(name:String, surname : String, dateOfBirth: Date, phoneNumber : String, email : String, gender: Gender, address : Address){
    this.name = name;
    this.surname = surname;
    this.dateOfBirth = dateOfBirth;
    this.phoneNumber = phoneNumber;
    this.email = email;
    this.gender = gender;
    this.address = address;
  }
}
